### PR TITLE
Work to fix the CI (failing related to scipy >= 1.12.0)

### PR DESCRIPTION
**Pull request by paugier**
Pyfftw tests are broken with scipy >= 1.12.0. I can reproduce locally with Python 3.9 and scipy >= 1.12.0. The tests pass with scipy == 1.11.4.

Last successful CI was run in 5 months ago: https://github.com/pyFFTW/pyFFTW/actions

```
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestDSTFloat32::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestIDSTFloat32::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestDSTNFloat32::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestIDSTNFloat32::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTNTestDSTNFloat32::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestDSTFloat64::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestIDSTFloat64::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestDSTNFloat64::test_normalized - AssertionError: False is not true
FAILED tests/test_pyfftw_scipy_fft.py::InterfacesScipyR2RFFTTestIDSTNFloat64::test_normalized - AssertionError: False is not true
```

Currently I don't understand the difference between scipy >= 1.12.0 and pyfftw and therefore I don't know how to fix pyfftw's CI.

The goal of this PR is to trigger the CI to show that it is broken without any changes in pyfftw code.

### CODE REVIEW

**Comment by paugier** [comment]
The failure could be related to these changes in Scipy:

- https://github.com/scipy/scipy/pull/18613 (issue https://github.com/scipy/scipy/issues/18452)

**Comment by paugier** [comment]
I fixed the issue with Scipy by following the new behavior.

Now there is another issue with building the wheel on Macos and a problem with PyPy on Windows.

**Comment by paugier** [comment]
Issue with **repairing** the wheel on macos...

```
   Fixing: /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/cibw-run-it7krc32/pp39-macosx_x86_64/built_wheel/pyFFTW-0.13.1+25.gd87a77b-pp39-pypy39_pp73-macosx_10_15_x86_64.whl
  Traceback (most recent call last):
    File "/private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/cibw-run-it7krc32/pp39-macosx_x86_64/build/venv/bin/delocate-wheel", line 8, in <module>
      sys.exit(main())
    File "/private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/cibw-run-it7krc32/pp39-macosx_x86_64/build/venv/lib/pypy3.9/site-packages/delocate/cmd/delocate_wheel.py", line 110, in main
      copied = delocate_wheel(
    File "/private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/cibw-run-it7krc32/pp39-macosx_x86_64/build/venv/lib/pypy3.9/site-packages/delocate/delocating.py", line 1004, in delocate_wheel
      out_wheel_fixed = _check_and_update_wheel_name(
    File "/private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/cibw-run-it7krc32/pp39-macosx_x86_64/build/venv/lib/pypy3.9/site-packages/delocate/delocating.py", line 839, in _check_and_update_wheel_name
      raise DelocationError(
  delocate.libsana.DelocationError: Library dependencies do not satisfy target MacOS version 10.15:
  /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpbg2r9jov/wheel/pyfftw/.dylibs/libfftw3f_threads.3.dylib has a minimum target of 12.0
  /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpbg2r9jov/wheel/pyfftw/.dylibs/libfftw3l.3.dylib has a minimum target of 12.0
  /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpbg2r9jov/wheel/pyfftw/.dylibs/libfftw3f.3.dylib has a minimum target of 12.0
  /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpbg2r9jov/wheel/pyfftw/.dylibs/libfftw3_threads.3.dylib has a minimum target of 12.0
  /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpbg2r9jov/wheel/pyfftw/.dylibs/libfftw3.3.dylib has a minimum target of 12.0
  /private/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpbg2r9jov/wheel/pyfftw/.dylibs/libfftw3l_threads.3.dylib has a minimum target of 12.0
```

**Comment by hgomersall** [comment]
That's great work, well done! I suggest we deal with the Mac stuff separately.

File: pyfftw/interfaces/scipy_fftpack.py [hunk 1]
```diff
@@ -580,12 +580,10 @@ def _dst(x, type=2, n=None, axis=-1, norm=None, overwrite_x=False,
                          '"ortho" or "forward".')
 
     if type == 3 and norm == 'ortho':
-        x = numpy.copy(x)
+        x = numpy.copy(x) / math.sqrt(2*x.shape[axis])
```
Comment (hgomersall) [inline pyfftw/interfaces/scipy_fftpack.py:583]: Does this result in an additional copy? Should we do it explicitly in-place (`x /= ...`)?

**Comment by hgomersall** [comment]
There is one comment in the code which I'm not entirely sure about. Beyond that I think trusting the scipy implementation is good enough.

**Comment by hgomersall** [comment]
So close! Just one failure on windows pypy that superficially looks tractable.

File: pyfftw/interfaces/scipy_fftpack.py (hunk 1 above)
Comment (paugier) [reply pyfftw/interfaces/scipy_fftpack.py]: I changed this to use `/=`!

**Comment by hgomersall** [comment]
Great work, thank you for pursuing this!

**Review by hgomersall — APPROVED** [review]

**Pull request merged**